hdat/hdat_cli.py

```diff
@@ -113,3 +124,90 @@ def diff_results(suites, golden_result, result):
         traceback.print_exc()
         msg = 'Error when attempting to show "{}": {}'
         raise AbortError(msg.format(print_resultspec(result), e))
+
+
+def get_result_keys(result, input_key_list):
+    keys = []
+
+    for in_key in input_key_list:
+        if ".*" in in_key:
+            nested_in_key = in_key.replace(".*", "")
+            for result_key in result.keys():
+                if nested_in_key in result_key and isinstance(result[result_key], dict):
+                    for nested_key in result[result_key].keys():
+                        keys.append(".".join([nested_in_key, nested_key]))
+        elif in_key in result.keys():
+            keys.append(in_key)
+    return sorted(keys)
+
+
+def get_unused_keys(input_key_list, distinct_keys):
+    unused_keys = list(set(input_key_list)-set(distinct_keys))
+    for undefined_key in unused_keys:
+        if ".*" in undefined_key:
+            base_key, wildcard = undefined_key.split(".")
```

Comment:
If the second term of the split isn't needed, you could put:
`base_key, _ = undefined_key.split(".")` instead.
This keeps any unused variable warnings from coming up and reduces confusion (a similar thing can be done if you don't actually use something from an iterable like `for _ in list_of_things`)

Reply:
Great! I did not know about this; I just fixed it!